js/src/hooks/useTour.js

```diff
@@ -0,0 +1,54 @@
+/**
+ * External dependencies
+ */
+import { useCallback, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useAppDispatch } from '.~/data';
+import useAppSelectDispatch from '.~/hooks/useAppSelectDispatch';
+
+/**
+ * @typedef {import('.~/data/selectors').Tour} Tour
+ */
+
+/**
+ * @typedef {Object} TourHook
+ * @property { { data: Tour|null, hasFinishedResolution: boolean, isResolving: boolean, invalidateResolution: Function } } tour The tour object
+ * @property {() => Promise<{tour: Tour}>} closeTour Closes the tour and sets it as checked.
+ * @property {boolean} showTour Indicates if the tour should be shown based on it's checked prop
+ */
+
+/**
+ * This hook returns a tour status based on a Tour ID, a setter for the tour and a property indicating if the tour should be shown.
+ *
+ * @param {string} tourId Tour ID to get
+ * @return {TourHook} The tour Hook
+ */
+const useTour = ( tourId ) => {
+	const tour = useAppSelectDispatch( 'getTour', tourId );
+	const { upsertTour } = useAppDispatch();
+	const [ tourClosed, setTourClosed ] = useState( false );
+	const checked = tour.data?.checked;
+	const showTour = tour.hasFinishedResolution && ! checked && ! tourClosed;
+
+	const closeTour = useCallback( () => {
+		const nextTour = { id: tourId, checked: true };
+
+		if ( ! checked ) {
+			setTourClosed( true );
+			return upsertTour( nextTour );
+		}
+
+		return Promise.resolve( { tour: nextTour } );
+	}, [ checked, setTourClosed, upsertTour, tourId ] );
+
+	return {
+		tour,
+		closeTour,
+		showTour,
```

Comment:
`closeTour` is a function and `showTour` is a boolean, the first time that I read the code I thought that `closeTour` and `showTour` were both functions, can we change `showTour` for something like `isTourShown` or something like that?

Reply:
For me no problem

Reply:
> `closeTour` is a function and `showTour` is a boolean, the first time that I read the code I thought that `closeTour` and `showTour` were both functions, can we change `showTour` for something like `isTourShown` or something like that?

These return values no longer exist after changing to use the alternative solution. Ref: https://github.com/woocommerce/google-listings-and-ads/pull/1879/commits/2aa7988850f990b2480251e42d8c111772bb71fe